### PR TITLE
Save rating in state

### PR DESCRIPTION
**Pull request by emilyb7**
ratings created are now saved in the state and shown on the view-goal page

### CODE REVIEW

File: app/components/feedback/feedback.jsx [hunk 1]
```diff
@@ -2,28 +2,29 @@ import React from 'react';
...
-      />
-    <div className="feedback-buttonContainer goal-detail-buttonContainer">
-      <div className="button-outer">
-        <button
-          type="button"
-          name="button"
-          className="new-feedback-button"
-          >Save</button>
+const Feedback = ({ currentGoal, onInputFeedback, saveRating }) => (
+    <div className='feedback-page'>
```
Comment (des-des) [inline app/components/feedback/feedback.jsx]: OK it look like this whole piece of diff is caused by tab. Now is double tabbed

File: app/components/view-goal/progress-bar.jsx [hunk 2]
```diff
@@ -1,22 +1,23 @@
 import React from 'react';
 
-const ProgressBarComponent = ({ progress, time }) => {
+const ProgressBarComponent = ({ rating, }) => {
+  const score = parseInt(rating.score);
```
Comment (des-des) [inline app/components/view-goal/progress-bar.jsx]: Check mdn for def of parseInt. You need a second argument

File: app/reducers/index.js [hunk 3]
```diff
@@ -9,6 +9,27 @@ const defaultState = {
...
+export const saveRating = (state, time, id) => {
+  const newRating = {
+    score: state.currentGoal.newRating.score,
+    id: id,
+    time: time,
+    comment: state.currentGoal.newRating.comment,
+  };
+  const currentGoal = {
+    ...state.currentGoal,
+    ratings: [newRating].concat((state.currentGoal.ratings || [])),
```
Comment (des-des) [inline app/reducers/index.js]: I think we need to increment the update count here.. I would make sure that you have a single function that is responsible for changing a goal. Feel free to talk to me about this!

File: app/reducers/index.js [hunk 4]
```diff
@@ -89,6 +110,12 @@ export default (state = defaultState, action) => {
           },
         },
       }
+    case types.SAVE_RATING:
+      return {
+        ...saveRating(state, action.time, action.id),
```
Comment (des-des) [inline app/reducers/index.js]: If your reducers cases are each held in their own function, then things work nicer here. You do not need to change anything, but ask me about this.

File: tests/reducer-test.js [hunk 5]
```diff
@@ -179,6 +181,69 @@ tape('test reducer MOVE_SLIDER: new rating added to currentGoal obj', (t) => {
...
+  const newState = {
+    ...initialState,
+    currentGoal: {
+      ...initialState.currentGoal,
+      newRating: {},
+      ratings: [ {score: 5, comment: "hello", id: 0, time: "yesterday",}, ],
+    },
+    goals: [{
+      ...myGoal,
+      ratings: [ {score: 5, comment: "hello", id: 0, time: "yesterday",}, ],
```
Comment (des-des) [inline tests/reducer-test.js]: Trailing commas everywhere! Chose a style then enforce it with a linter.

Comment (emilyb7) [inline tests/reducer-test.js]: linter installed in #180

**Comment by des-des** [comment]
@emilyb7 reassign to me when conflicts are fixed!

**Pull request merged**